Library/Source/RuntimeImplV8.cpp

```diff
@@ -0,0 +1,78 @@
+#include "RuntimeImpl.h"
+
+#include <napi/env.h>
+
+#include <v8.h>
+#include <libplatform/libplatform.h>
+
+namespace Babylon
+{
+    namespace
+    {
+        class Module final
+        {
+        public:
+            Module(const char* executablePath)
+            {
+                v8::V8::InitializeICUDefaultLocation(executablePath);
+                v8::V8::InitializeExternalStartupData(executablePath);
+                m_platform = v8::platform::NewDefaultPlatform();
+                v8::V8::InitializePlatform(m_platform.get());
+                v8::V8::Initialize();
+            }
+
+            ~Module()
+            {
+                v8::V8::Dispose();
+                v8::V8::ShutdownPlatform();
+            }
+
+            static void Initialize(const char* executablePath)
+            {
+                if (s_module == nullptr)
+                {
+                    s_module = std::make_unique<Module>(executablePath);
+                }
+            }
+
+        private:
+            std::unique_ptr<v8::Platform> m_platform;
+
+            static std::unique_ptr<Module> s_module;
+        };
+
+        std::unique_ptr<Module> Module::s_module;
+
+        v8::Isolate* CreateIsolate(const char* executablePath)
+        {
+            Module::Initialize(executablePath);
+
+            v8::Isolate::CreateParams create_params;
+            create_params.array_buffer_allocator = v8::ArrayBuffer::Allocator::NewDefaultAllocator();
+            return v8::Isolate::New(create_params);
+        }
+
+        void DestroyIsolate(v8::Isolate* isolate)
+        {
+            // todo : GetArrayBufferAllocator not available?
+            //delete isolate->GetArrayBufferAllocator();
+            isolate->Dispose();
+        }
```

Comment:
not sure these helper function are useful anymore. perhaps inline them in BaseThreadProcedure?